feeds/wlan-ap/opensync/src/src/sm/src/ubus_collector.c

```diff
@@ -0,0 +1,668 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+
+#include "ubus_collector.h"
+
+/* Global list of events received from hostapd */
+dpp_event_report_data_t g_report_data;
+
+/* Internal list of processed events ready to be deleted from hostapd */
+static dpp_event_report_data_t *deletion_pending = NULL;
+static struct ubus_context *ubus = NULL;
+static char *ubus_object = NULL;
+
+typedef struct {
+	uint64_t session_id;
+	char *bss;
+
+	ds_dlist_node_t node;
+} delete_entry_t;
+
+static int client_first_data_event_cb(struct blob_attr *msg,
+				      dpp_event_record_session_t *dpp_session,
+				      uint64_t event_session_id)
+{
+	int error = 0;
+	struct blob_attr *tb_client_first_data_event[__CLIENT_FIRST_DATA_MAX] = {
+		0
```

Comment:
this will only 0 the very first element in the struct. If you want an implicit memset it needs to be {}